docs/tls/CONFIGURATION.md

```diff
@@ -0,0 +1,183 @@
+# Configuring NLK to use TLS / mTLS guide
+
+This guide describes the steps required to configure NLK to use self-signed certificates for TLS / mTLS.
```

Comment:
Oh boy. I both love and despise this idea. Given Brian's stand that each document should stand on its own, this would mean a ton of duplication / super-similar content. I am more of a bite-sized documentation person than a dump-it-all-at-once person, so I like the idea that way.

Bottom line, I have to get the code written at some point, so I am going to pretty much go ahead with this the way it is. The feedback from you and Brian has really improved the documentation, and is much appreciated.

Reply:
Now, I am not about one big huge doc.  But discrete docs need to give the reader enough of the clue that they should not have to go back to the root doc to have context.  If each doc stands alone, it is complete in itself and what it is describing.
You should not interpret that as duplicating a bunch of context.
Go here for generating self-signed certs, go here for configuring TLS, go here for configuring mTLS, and so forth.
Phrased differently, don't over-rotate.
The idea is that a doc about generating self-signed certs should be able to be used by multiple referring docs.  If not, then it should not be a separate doc.